pilot/pkg/networking/plugin/authz/rbac_v2.go

```diff
@@ -0,0 +1,171 @@
+		return &http_config.RBAC{Rules: rbac}
+	}
+	// Get all AuthorizationPolicy Istio config from this namespace.
+	allAuthzPolicies := option.authzPolicies.NamespaceToAuthorizationConfigV2[namespace].AuthzPolicies
+	// Get all key-value pIstioairs of AuthorizationPolicy name to list of ServiceRoleBindings in this namespace.
+	for _, authzPolicy := range allAuthzPolicies {
+		// If a WorkloadSelector is used in the AuthorizationPolicy config and does not match this service, skip this
+		// AuthorizationPolicy.
+		if authzPolicy.Policy.WorkloadSelector != nil &&
+			!(model.LabelsCollection{service.labels}.IsSupersetOf(authzPolicy.Policy.WorkloadSelector.Labels)) {
+			continue
+		}
+		for i, binding := range authzPolicy.Policy.Allow {
+			// Check if we have converted this ServiceRole before.
+			permissions := make([]*policyproto.Permission, 0)
+			if !hasConvertedPermission(rbac, binding.RoleRef.Name) {
+				// Convert role.
+				role := option.authzPolicies.RoleForNameAndNamespace(namespace, binding.RoleRef.Name)
+				for _, rule := range role.Rules {
+					// Check to make sure the services field does not exist in the ServiceRole.
+					if len(rule.Services) > 0 {
+						rbacLog.Errorf("should not have services field in ServiceRole (found in %s)", binding.RoleRef.Name)
+						continue
+					}
+					// Check to make sure that the current service (caller) is the one this rule is applying to.
+					if !service.areConstraintsMatched(rule) {
+						continue
+					}
+					if option.forTCPFilter {
+						// TODO(yangminzhu): Move the validate logic to push context and add metrics.
+						if err := validateRuleForTCPFilter(rule); err != nil {
+							// It's a user misconfiguration if a HTTP rule is specified to a TCP service.
+							// For safety consideration, we ignore the whole rule which means no access is opened to
+							// the TCP service in this case.
+							rbacLog.Debugf("rules[%d] ignored, found HTTP only rule for a TCP service: %v", i, err)
+							continue
+						}
+					}
+					// Generate the policy if the service is matched and validated to the services specified in
+					// ServiceRole.
+					permissions = append(permissions, convertToPermission(rule))
+				}
+				if len(permissions) == 0 {
+					rbacLog.Debugf("role %s skipped for no rule matched", binding.RoleRef.Name)
+					continue
+				}
+			}
+
+			if option.forTCPFilter {
+				if err := validateBindingsForTCPFilter([]*rbacproto.ServiceRoleBinding{binding}); err != nil {
+					rbacLog.Debugf("role %s skipped, found HTTP only binding for a TCP service: %v", binding.RoleRef.Name, err)
+					continue
+				}
+			}
+			// Convert binding.
+			enforcedPrincipals, _ := convertToPrincipals([]*rbacproto.ServiceRoleBinding{binding}, option.forTCPFilter)
+
+			if len(enforcedPrincipals) == 0 {
+				rbacLog.Debugf("role %s skipped for no principals found", binding.RoleRef.Name)
+				continue
+			}
+
+			policyName := "authz-policy-" + authzPolicy.Name + "-allow-" + strconv.Itoa(i)
```

Comment:
fmt.sprintf("authz-policy-%s-allow[%d]", authzPolicy.Name, i)

Reply:
Thank you for the suggestion!